code/FBX/FBXConverter.cpp

```diff
@@ -3522,54 +3549,27 @@ void FBXConverter::SetShadingPropertiesRaw(aiMaterial* out_mat, const PropertyTa
             out->mMetaData->Set(14, "CustomFrameRate", doc.GlobalSettings().CustomFrameRate());
         }
 
-        void FBXConverter::ConvertToUnitScale( FbxUnit unit ) {
-            if (mCurrentUnit == unit) {
-                return;
-            }
-
-            ai_real scale = 1.0;
-            if (mCurrentUnit == FbxUnit::cm) {
-                if (unit == FbxUnit::m) {
-                    scale = (ai_real)0.01;
-                } else if (unit == FbxUnit::km) {
-                    scale = (ai_real)0.00001;
-                }
-            } else if (mCurrentUnit == FbxUnit::m) {
-                if (unit == FbxUnit::cm) {
-                    scale = (ai_real)100.0;
-                } else if (unit == FbxUnit::km) {
-                    scale = (ai_real)0.001;
-                }
-            } else if (mCurrentUnit == FbxUnit::km) {
-                if (unit == FbxUnit::cm) {
-                    scale = (ai_real)100000.0;
-                } else if (unit == FbxUnit::m) {
-                    scale = (ai_real)1000.0;
-                }
+        ai_real FBXConverter::GetUnitScale( FbxUnit unit )
+        {
+            double scale;
+            if( out->mMetaData->Get("UnitScaleFactor", scale) )
+            {
+                printf("unit scale factor from file: %f\n", scale);
             }
-            
-            for (auto mesh : meshes) {
-                if (nullptr == mesh) {
-                    continue;
-                }
-
-                if (mesh->HasPositions()) {
-                    for (unsigned int i = 0; i < mesh->mNumVertices; ++i) {
-                        aiVector3D &pos = mesh->mVertices[i];
-                        pos *= scale;
-                    }
-                }
+            else
+            {
+                printf("no file scale detected!");
```

Comment:
Logging without error. Isn't this common?

Reply:
If no file scale exists, doesn't make sense to error and not read the file, we can assume the scale is 1.0.

Reply:
@kimkulling do you have any other way of providing feedback to user which isn't an error, or should we just remove the prints?

Reply:
Hi, sure, just use **ASSIMP_LOG_DEBUG** for this instead of printf.

Reply:
Thanks!